src/components/Select/Select.tsx

```diff
@@ -39,8 +39,6 @@ export interface SelectGroup {
 export interface BaseProps {
   /** List of options or option groups to choose from */
   options?: (SelectOption | SelectGroup)[];
```

Comment:
I'm not positive but I this might require more work, was the intention to have this
```suggestion
  options?: SelectOption[];
```
?

Reply:
And if it was we may want to deprecate the type

Reply:
We still want the ability to specify groups which render as `<optgroup>` elements within the `select` in the html, we just don't want to have two props when the options prop can handle both cases. @ry5n you added this deprecation in https://github.com/Shopify/polaris-react-deprecated/pull/1576, can you confirm that was the intent?

Stick the following in your Playground (yay having a function based playground is a nice little thing):


```js
import React from 'react';
import {Page, Select} from '../src';

export default function Playground() {
  return (
    <Page title="Playground">
      <Select
        label="Date range"
        options={[
          {label: 'Today', value: 'today'},
          {label: 'Yesterday', value: 'yesterday'},
          {label: 'Last 7 days', value: 'lastWeek'},
        ]}
        onChange={() => {}}
        value="today"
      />
      <Select
        label="Date range"
        options={[
          {
            title: 'First',
            options: [
              {label: 'Today', value: 'today'},
              {label: 'Yesterday', value: 'yesterday'},
              {label: 'Last 7 days', value: 'lastWeek'},
            ],
          },
          {
            title: 'Second',
            options: [
              {label: 'Today', value: 'today'},
              {label: 'Yesterday', value: 'yesterday'},
              {label: 'Last 7 days', value: 'lastWeek'},
            ],
          },
        ]}
        onChange={() => {}}
        value="today"
      />
    </Page>
  );
}
```

Reply:
The idea was exactly that: not needing 2 props for grouped and ungrouped options. One thing I did badly was not checking to see what other lists still do it with 2 props. I think OptionList and ChoiceList might, so I might have made more work/an inconsistent API with that choice 😭